File: JS/FindSmallestLetterGreaterThanTarget.test.js
/**
 * @version JAVASCRIPT ECMAScript 6 
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { nextGreatestLetter } from './FindSmallestLetterGreaterThanTarget.js'

TEST('Starting FindSmallestLetterGreaterThanTarget test...', (t) => {
    //Test 1
    let want = "c"
    let got = nextGreatestLetter( ["c","f","j"], "a" )
    t.assert( want === got, "Expect: "+want)
    //Test 2
    want = "f"
    got = nextGreatestLetter( ["c","f","j"], "c" )
    t.assert( want === got, "Expect: "+want)
    //Test 3
    want = "x"
    got = nextGreatestLetter( ["x","x","y","y"], "z" )
    t.assert( want === got, "Expect: "+want)

    t.end()
})

File: JAVA/FindSmallestLetterGreaterThanTargetTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class FindSmallestLetterGreaterThanTargetTest {

    @Test
	public void testFindSmallestLetterGreaterThanTarget() throws Exception {
        //Test 1
        char want = 'c';
        char got = FindSmallestLetterGreaterThanTarget.nextGreatestLetter( new char[]{'c', 'f', 'j'}, 'a' );
        assertEquals(want, got);
        //Test 2
        want = 'f';
        got = FindSmallestLetterGreaterThanTarget.nextGreatestLetter( new char[]{'c', 'f', 'j'}, 'c' );
        assertEquals(want, got);
        //Test 3
        want = 'x';
        got = FindSmallestLetterGreaterThanTarget.nextGreatestLetter( new char[]{'x', 'x', 'y', 'y'}, 'z' );
        assertEquals(want, got);
	}

}
